File: bean-registrar/src/test/java/com/example/demo/ApplicationTest.java
package com.example.demo;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;

import static org.assertj.core.api.Assertions.assertThat;

@SpringJUnitConfig(CustomConfig.class)
class ApplicationTest {

    @Autowired
    PostRepository postRepository;

    @Test
    void testPostRepository() {
        assertThat(this.postRepository).isInstanceOf(InMemoryPostRepository.class);
    }
}

File: bean-registrar-kotlin-dsl/src/test/kotlin/com/example/demo/ApplicationTest.kt
package com.example.demo

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.junit.jupiter.SpringJUnitConfig

@SpringJUnitConfig(CustomConfig::class)
class ApplicationTest @Autowired constructor(private val postRepository: PostRepository) {

    @Test
    fun testPostRepository() {
        assertThat(this.postRepository).isInstanceOf(InMemoryPostRepository::class.java)
    }
}